packages/framework/ai-collab/src/aiCollabApi.ts

```diff
@@ -182,3 +186,76 @@ export interface TokenLimits {
 	 */
 	readonly outputTokens?: number;
 }
+
+/**
+ * Represents a path through a tree of objects.
+ * number values represent array indices whereas string values represent object keys.
+ *
+ * @alpha
+ */
+export type ObjectPath = (string | number)[];
+
+/**
+ * Represents a create operation between two branches of a tree.
+ * Meaning that an attribute (a shared tree node) was identified as being created.
+ *
+ * @alpha
+ */
+export interface DifferenceCreate {
+	type: "CREATE";
+	path: ObjectPath;
+	value: unknown;
+}
+
+/**
+ * Represents a remove operation between two branches of a tree.
+ * Meaning that an attribute (a shared tree node) was identified as being deleted.
+ * When using object ids, removes are idenitified by an object with a given id no longer existing.
+ *
+ * @alpha
+ */
+export interface DifferenceRemove {
+	type: "REMOVE";
+	path: ObjectPath;
+	oldValue: unknown;
```

Comment:
I'm not sure an old value makes sense for a remove operation; this means removing a tree node within an array. I would remove this.

Reply:
The UI will show what value have been deleted, so the old value needs to be set here.